src/hevm/src/EVM/SymExec.hs

```diff
@@ -583,14 +572,9 @@ equivalenceCheck solvers bytecodeA bytecodeB opts signature' = do
             (Revert _ a, Revert _ b) -> if a==b then PBool False else a ./= b
             (Revert _ _, _) -> PBool True
             (_, Revert _ _) -> PBool True
-            (Invalid _, Invalid _) -> PBool False
-            (Invalid _, _ ) -> PBool True
-            (_, Invalid _) -> PBool True
-            (EVM.Types.StackLimitExceeded _, EVM.Types.StackLimitExceeded _ ) -> PBool False
-            (EVM.Types.StackLimitExceeded _, _) -> PBool True
-            (_, EVM.Types.StackLimitExceeded _) -> PBool True
-            (a, b) -> if a == b then PBool False
-                                else error $ "Unimplemented, see TODO about EVM failures and TmpExpr. Left: " <> show a <> " Right: " <> show b
+            (Failure erra _, Failure errb _) -> if erra==errb then PBool False else PBool True
```

Comment:
This has slightly different semantics than the previous equivalence check. Previously, if end results were different and both of them were one of the unhandled error cases, then it would raise the unimplemented exception. Now all error are handled in the same way. Note that `TmpError` cannot happen here since `flattenExpr` catches it and raises an exception.

Reply:
This looks correct to me. However, for the sake of defensive coding, I'd try to check for `TmpError` here, and throw an `error`. Maybe in the future someone will use this function for something else, where `TmpError` hasn't been filtered yet.

Reply:
Do we want to return True if both programs return the same TmpError as before? Or raise an error if there is an unhandled error in either of the two programs, regardless of wether the error is the same?

Reply:
I'd just raise an `error.` I think `TmpError` is effectively an internal error, rather than a EVM error.

Reply:
I agree! Fixed